### PR TITLE
Use TextLine objects instead of the full plain text in coordinate extraction

### PR DESCRIPTION
**Pull request by stijnvermeeren-swisstopo**
- Separate the creation of TextLine object for a given PDF page into a separate method, that is used by both layer extraction and coordinate extraction.
- Instead of finding the coordinate key in the plain document text, find it in the TextLine objects, so that we also have the information of where it is located on the PDF page
- Instead of looking for coordinates within 100 characters of the coordinate key (in the defaut "reading order"), look for coordinates to the right and/or immediately below the coordinate key. This should be a little more robust / less dependent on they way (in which order) text is defined in the document.
- Instead of concatenating the text of all pages at the beginning of the coordinate extraction, we iterate over the pages and return on the first page where a valid coordinate was found. (This also will allow us to know on which page the coordinate was found, which is necessary for a future visualisation.)
- Improve logging a little bit.

The extraction of the coordinates values themselves still happens from plain text, not from the TextLine objects. We could improve on this in a future implementation, to use the TextLine objects everywhere. The main advantage of this would be, that we would also know where on the PDF page the coordinates were found, and we could use this information for the visualisation.

There are no changes in the KPIs for both the Zurich dataset and the Geoquat validation dataset.

### CODE REVIEW

File: src/stratigraphy/util/coordinate_extraction.py [hunk 1]
```diff
@@ -247,47 +250,51 @@ def extract_coordinates(self) -> Coordinate | None:
...
-            else:
-                coordinate = LV03Coordinate(
-                    CoordinateEntry(east),
-                    CoordinateEntry(north),
-                )
-            if coordinate.is_valid():
-                return coordinate
-
-        logger.warning(f"Could not extract valid coordinates from {coordinate_values}")
+            text = page.get_text()
```
Comment (redur) [inline src/stratigraphy/util/coordinate_extraction.py:254]: This comment relates to the the docstring of the function:

Suggestion:
```Extracts the coordinates from a borehole profile.

Processes the borehole profile page by page and tries to find the coordinates in the respective text of the page.
```

File: src/stratigraphy/util/extract_text.py
Comment (redur) [inline src/stratigraphy/util/extract_text.py:1]: I am not sure whether it makes sense to create a new python file for this. Could we thematically group it to line.py as this is the code that creates the objects defined in line.py?

File: src/stratigraphy/extract.py
Comment (redur) [inline src/stratigraphy/extract.py:1]: I am fine with that change. I am just thinking if there is any use case where we'd need the page / doc object inside process page. But should that be the case we can always add the page object again.

**Comment by redur** [comment]
This comes in handy with the changes I am thinking about for label studio with regards to coordinate extraction.

**Review by redur — APPROVED** [review]
Approved with some open questions.

File: src/stratigraphy/extract.py
Comment (stijnvermeeren-swisstopo) [reply src/stratigraphy/extract.py:1]: Good point, but at this point in time, I don't see a need for it.

Also, it feels more consistent now with the `geometric_lines`, which are derived from the page object in a similar way, but were already passed as a separate parameter to the `process_page` method.

File: src/stratigraphy/util/extract_text.py
Comment (stijnvermeeren-swisstopo) [reply src/stratigraphy/util/extract_text.py:1]: In general I'm more in favor of more files with fewer lines of code per file, and maybe group files into subdirectories/packages/modules when we start having too many files in one place.
But if you feel strongly about it, we could also put these in one file, they are certainly closely related.

Comment (redur) [reply src/stratigraphy/util/extract_text.py:1]: Follow up ticket regarding this is created on Jira.

**Pull request merged**